cli/init/index.js

```diff
@@ -1,50 +1,82 @@
 #!/usr/bin/env node
 
 (async () => {
-  const { constants, promises } = require('fs')
-  const { access, mkdir, readFile, writeFile } = promises
+  const { promises, existsSync } = require('fs')
+  const { mkdir, readFile, writeFile } = promises
   const fileOptions = { encoding: 'utf8' }
   const dir = "src"
   const beagleContent = await readFile(__dirname + '/boilerplate/beagle-service.ts')
   const appContent = await readFile(__dirname + '/boilerplate/app.tsx')
+  const appTsx = 'app.tsx'
+  const beagleServicePath = 'beagle/beagle-service.ts'
   const readline = require('readline')
   const rl = readline.createInterface({
     input: process.stdin,
     output: process.stdout,
   })
 
   await mkdir(`${dir}/beagle/`, { recursive: true });
-  await writeFile(`${dir}/beagle/beagle-service.ts`, beagleContent, fileOptions)
 
-  const createAppTsx = async (path, content) => {
+  const beagleServiceExists = async (path) => {
+    const exists = await existsSync(`${path}/${beagleServicePath}`)
+    return exists
+  }
+
+  const appTsxExists = async (path) => {
+    const exists = await existsSync(`${path}/app.tsx`)
+    if (exists) {
+      overwriteAppTsx()
+    }
+    else {
+      generateAppTsx()
+    }
+  }
+
+  const generateAppTsx = async () => {
+    await createFile(`${dir}/${appTsx}`, appContent)
+    console.log("success! all configuration files were created correctly")
+    process.exit()
+  }
+
+  const createFile = async (path, content) => {
     try {
-      await writeFile(`${path}/app.tsx`, content)
-      process.exit()
+      await writeFile(path, content, fileOptions)
     } catch (error) {
       console.error(error)
       process.exit(1)
     }
   }
-  const appTsxExists = async (path) => {
-    try {
-      await access(`${path}/app.tsx`, constants.F_OK)
-      return true
-    } catch (error) {
-      console.error("app.tsx did not exist, it was created with Beagle settings.")
-      return false
-    }
+
+  const overwriteAppTsx = async () => {
+    rl.question('Do you want to replace "app.tsx" content with the Beagle configuration (y or n)?', async (answer) => {
+      if (new RegExp(`^${answer}$`, 'i').test('y')) {
+        if (`${answer}` === 'y' || `${answer}` === 'Y') {
+          await createFile(`${dir}/app.tsx`, appContent)
+          console.log("success! all configuration files were created correctly")
+          process.exit()
+        }
+      }
```

Comment:
```suggestion
      if (new RegExp(`^${answer}$`, 'i').test('y')) {
         await createFile(`${dir}/app.tsx`, appContent)
         console.log("success! all configuration files were created correctly")
         process.exit()
      }
```
We have an unnecessary `if` here